lib/src/obsoleted/analyzer_plugin/analyzer_plugin_utils.dart

```diff
@@ -106,6 +109,33 @@ plugin.AnalysisErrorFixes metricReportToAnalysisErrorFixes(
       hasFix: false,
     ));
 
+Iterable<plugin.AnalysisErrorFixes> checkDeprecatedConfigOptions(
+  AnalyzerPluginConfig config,
+  Iterable<DeprecatedOption> deprecatedOptions,
+  String analysisOptionPath,
+) {
+  final ids = {
+    ...config.codeRules.map((rule) => rule.id),
+    ...config.methodsMetrics.map((metric) => metric.id),
+    ...config.antiPatterns.map((pattern) => pattern.id),
+    ...config.metricsConfig.keys,
+  };
+
+  return deprecatedOptions
+      .where((option) => ids.contains(option.deprecated))
+      .map((option) => plugin.AnalysisErrorFixes(plugin.AnalysisError(
+            plugin.AnalysisErrorSeverity.WARNING,
+            plugin.AnalysisErrorType.HINT,
+            plugin.Location(analysisOptionPath, 0, 1, 0, 0, 0, 0),
```

Comment:
It seems that it should extend current message with deprecation warning instead of creating a new one on top of a file.

Reply:
fixed